### PR TITLE
372: Tool support for small/minor/trivial "fast-track" changes.

### PR DESCRIPTION
**Pull request by rwestberg**
Hi all,

Please review this change that adds the possibility of issuing an `/issue create` command. The title and the body of the issue will be copied from the pull request. The command is only available for committers and above (who already have access to creating issues manually).

Best regards,
Robin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-372](https://bugs.openjdk.java.net/browse/SKARA-372): Tool support for small/minor/trivial "fast-track" changes.

### Reviewers
 * Erik Helin ([ehelin](@edvbld) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/677/head:pull/677`
`$ git checkout pull/677`

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java [hunk 1]
```diff
@@ -98,6 +107,170 @@ private void showHelp(PrintWriter reply) {
...
+                var currentList = currentSolved.stream()
+                                               .map(id -> "`" + id + "`")
+                                               .collect(Collectors.joining(", "));
+                reply.println(currentList);
+            }
+        }
+    }
+
+    private void createIssue(PullRequestBot bot, PullRequest pr, String args, CensusInstance censusInstance, HostUser author, PrintWriter reply) {
+        if (!censusInstance.isAuthor(author)) {
```
[the remark this answers is not in the document]
Comment (rwestberg) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java:216]: No, I should have written Author in the PR description, think that check is sufficient as all authors have permissions to create issues manually in JBS.

[the remark this answers is not in the document]
Comment (edvbld) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java:216]: Ok!

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java [hunk 2]
```diff
@@ -98,6 +107,170 @@ private void showHelp(PrintWriter reply) {
...
+            priority = next.substring(1);
+            next = argSplit.pollFirst();
+        }
+
+        // Next argument is the mandatory component name
+        if (next == null) {
+            showHelp(reply);
+            return;
+        }
+        var component = next;
```
Comment (edvbld) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java:245]: Given that you incorporated my changes to the help text I think you need to add some code here to handle the case when `component` is something like `"hotspot/gc"` :smiley:

Comment (rwestberg) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java:245]: But it does work, there's even a test for it. :)

Comment (edvbld) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/IssueCommand.java:245]: Sigh, I didn't notice the regular expression in `args.split("(?:\\s+|/)"))` :sleeping: Looks good then :+1:

**Pull request closed without merging**